openmdao/components/tests/test_eq_constraint_comp.py

```diff
@@ -540,6 +540,43 @@ def test_renamed_vars(self):
 
         assert_check_partials(cpd, atol=1e-5, rtol=1e-5)
 
+    def test_multidimentional_shape(self):
```

Comment:
We should have an equivalent test for BalanceComp, and we should probably also test this with `normalize=False`.